memgpt/cli/cli.py

```diff
@@ -18,6 +22,74 @@
 from memgpt.constants import MEMGPT_DIR, CLI_WARNING_PREFIX
 from memgpt.agent import Agent
 from memgpt.embeddings import embedding_model
+from memgpt.server.constants import WS_DEFAULT_PORT, REST_DEFAULT_PORT
+
+
+class ServerChoice(Enum):
+    rest_api = "rest"
+    ws_api = "websocket"
+
+
+def server(
+    type: ServerChoice = typer.Option("rest", help="Server to run"), port: int = typer.Option(None, help="Port to run the server on")
+):
+    """Launch a MemGPT server process"""
+
+    if type == ServerChoice.rest_api:
+        if port is None:
+            port = REST_DEFAULT_PORT
+
+        # Change to the desired directory
+        script_path = Path(__file__).resolve()
+        script_dir = script_path.parent
+
+        server_directory = os.path.join(script_dir.parent, "server", "rest_api")
+        command = f"uvicorn server:app --reload --port {port}"
+
+        # Run the command
+        print(f"Running REST server: {command} (inside {server_directory})")
+
+        try:
+            # Start the subprocess in a new session
+            process = subprocess.Popen(command, shell=True, start_new_session=True, cwd=server_directory)
+            process.wait()
+        except KeyboardInterrupt:
+            # Handle CTRL-C
+            print("Terminating the server...")
+            process.terminate()
+            try:
+                process.wait(timeout=5)
+            except subprocess.TimeoutExpired:
+                process.kill()
+            sys.exit(0)
+
+    elif type == ServerChoice.ws_api:
+        if port is None:
+            port = WS_DEFAULT_PORT
+
+        # Change to the desired directory
+        script_path = Path(__file__).resolve()
+        script_dir = script_path.parent
+
+        server_directory = os.path.join(script_dir.parent, "server", "ws_api")
+        command = f"python server.py {port}"
```

Comment:
We could technically run the WS server process using python directory instead of a subprocess, like we do in the `server.py` file (for websockets):

```python
if __name__ == "__main__":
    port = WS_DEFAULT_PORT

    # Check if a port argument is provided
    if len(sys.argv) > 1:
        try:
            port = int(sys.argv[1])
        except ValueError:
            print(f"Invalid port number. Using default port {port}.")

    server = WebSocketServer(port=port)
    asyncio.run(server.run())
```

However I just used subprocess to match the REST call and in case we want to run these as background processes at a later point (eg if you want to spawn both servers at the same time) it might look cleaner if it's a subprocess instead of a direct python submodule.

IDK though can do it the other way too if you have a strong opinion @sarahwooders

Reply:
I think its better to avoid doing `python ...`, since if the user wants to run a background process they could just do `memgpt server &`?

Reply:
Yeah agreed. I'm going to leave as-is for now since we can actually run websockets via fastapi so we should probably do that instead (so it will be refactored again anyways).